docs/source/developer_guide/internals.md

```diff
@@ -1,11 +1,9 @@
-Internals
----------
+# Internals
```

Comment:
This doc seems to be only about GeoArrow, not all data structures used in cuSpatial. And 'Internals" is very broad. I think it should be more specifically named. Should it be renamed to geoarrow.md?

Reply:
Actually, i remember we discussed eliminating this from our docs and referencing an official GeoArrow spec.

Reply:
The idea is that geoarrow spec is centrally hosted at https://github.com/geopandas/geo-arrow-spec. The geoarrow part of this document can be removed and refer to that document. The rest of the document should be refactored into "library_design" document mentioned in #599.

cc. @thomcom

Reply:
btw, we should add `cuspatial` into the "implementation" section of `geo-arrow-spec` since #585 recently implemented it. Maybe even highlight it as "GPU backed" or something.

Reply:
fyi I'm more inclined to merge this PR first and punt refactoring to a later PR.